keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -174,7 +177,11 @@ class ResourceActuator(
         publisher.publishEvent(ResourceCheckUnresolvable(resource, e, clock))
       } catch (e: Exception) {
         log.error("Resource check for $id failed [$coid]", e)
-        publisher.publishEvent(ResourceCheckError(resource, e, clock))
+        val normalizedException = when (e) {
+          is UserException, is SystemException -> e
+          else -> SystemException(e)
+        } as SpinnakerException
+        publisher.publishEvent(ResourceCheckError(resource, normalizedException, clock))
```

Comment:
This ensures that the exception type of a `ResourceCheckError` will always be `UserException` or `SystemException`.